spec/Section 3 -- Type System.md

```diff
@@ -1923,3 +1929,62 @@ type ExampleType {
   oldField: String @deprecated(reason: "Use `newField`.")
 }
 ```
+
+### @defer
+
+```graphql
+directive @defer(label: String, if: Boolean) on FRAGMENT_SPREAD | INLINE_FRAGMENT
+```
+
+The `@defer` directive may be provided for fragment spreads and inline fragments to 
```

Comment:
shall we add a non-normative note here that the executor is allowed to ignore a defer directive if for instance the execution is more performant in one go. This can happen if for instance we defer a field that is anyway already in the context.

```graphql
{
   person {
      ... @defer {
         name
      }
}
```

Hot Chocolate for instance optimizes such defers. I know Facebook also does similar things in their implementation. I remember we discussed this in the WG and also in the initial proposal but recent comments of users made clear that the spec text is not clear on this fact.

Reply:
https://github.com/graphql/graphql-wg/blob/main/rfcs/DeferStream.md#server-requirements-for-defer-and-stream

Reply:
Even if it’s in the context, sending it prematurely will delay delivery of the first payload, even inasmuch as it will just make the first payload larger.

Lets say inclusion of the deferred field adds 5 ms, and deferral saves that 5 ms for the first payload but because of overhead causes the field to arrive 20 ms after the first payload.

a server might reasonably think it’s better to just send the field and a client might disagree!

It seems strange to automatically let the server win, sure the server has more information in terms of the actual time differences, but the client knows more about what it wants.

this seems like a great place for arguments for defer to specify how this should be handled.

so then the question becomes what the default should be. From a language semantics standpoint, the default for defer should seemingly be to always defer, but I understand from a practical standpoint perhaps it makes sense to go with the current implementation. But it doesn’t make me happy, seems to introduce some non specified behavior into the spec, which graphql has successfully avoided until now.

Reply:
In your experience, what is the actual overall delay added when deferring a field that is already in context? Is there really that much overhead. My numbers above are shots in the dark, it would be great to bring some real world data in for this decision

Reply:
Depends on the DataSet really but let's say we are on a list of a large dataset ... not doing the defers in the case will significantly save on time and on data. Each patch is a full GraphQL response.

Reply:
Ok, I guess I am left to wonder what the client is to do in such an example if they want to defer a few fields in such a large list because of delivery time itself, not because of execution time. If in a list myLongList  of 1,000,000 Item objects, we have two fields, field necessaryNow and couldBeLater, by including couldBeLater in the initial response, you are delaying delivery of necessaryNow.

The query could be constructed perhaps differently with duplicate references to myLongList, deferring one and separating the fields, but how is the client to really know if that disables the server “optimization”? Even if it works, it is certainly quite opaque. This seems like behavior that should be behind an argument.

Reply:
@michaelstaib I added the note to the bottom of this file. Also posted in the relevant discussion topic more background on why it's there https://github.com/robrichard/defer-stream-wg/discussions/7#discussioncomment-1672862

Reply:
Great that a better way to discuss things then spamming the PR :)